utils/urlx/urlx.go

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) Mondoo, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package urlx
+
+import (
+	"fmt"
+	"strings"
+)
+
+// ParseGitSshUrl retrieves the
+func ParseGitSshUrl(url string) (string, string, string, error) {
```

Comment:
Can we add a few tests so its easier to understand how this parses. eg. with github url, gitlab url and custom git url

Reply:
+1, done